File: commons-utils/src/main/java/io/github/cooperlyt/commons/data/Person.java
package io.github.cooperlyt.commons.data;

import lombok.Getter;

import java.time.LocalDateTime;

public interface Person extends PowerBody{

  @Getter
  enum Sex {
    MALE("男"),
    FEMALE("女");

    private final String label;

    Sex(String label) {
      this.label = label;
    }
  }

  Sex getSex();

  LocalDateTime getBirthday();

  String getNation();

  int getEthnicity();

  String getAddress();

  String getPhoto();



}

File: commons-utils/src/main/java/io/github/cooperlyt/commons/data/Person.kt
package io.github.cooperlyt.commons.data

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import java.time.LocalDateTime

@JsonSerialize(`as` = Person::class)
@JsonDeserialize(`as` = Person.Sample::class)
interface Person : PowerBody {

  enum class Sex(val label: String) {
    MALE("男"),
    FEMALE("女");
  }

  val sex: Sex

  val birthday: LocalDateTime?

  val nation: String?

  val ethnicity: Int?

  val address: String?

  val email: String?

  data class Sample(
    override val sex: Sex,
    override val birthday: LocalDateTime?,
    override val nation: String?,
    override val ethnicity: Int?,
    override val address: String?,
    override val email: String?,
    override val name: String,
    override val idType: IdentityType,
    override val idNumber: String,
    override val tel: String?
  ): Person
}